indexer-api/src/infra/storage/dust.rs

```
@@ -354,21 +359,24 @@ impl DustStorage for Storage {
 
                     let mut matching_prefixes = Vec::new();
                     for (nullifier_bytes,) in nullifiers {
-                        let nullifier: indexer_common::domain::DustNullifier =
-                            nullifier_bytes.as_slice().try_into().unwrap();
-                        let nullifier_hex = nullifier.hex_encode().to_string();
-                        for prefix in prefixes {
-                            let hex_prefix = prefix.hex_encode().to_string();
-                            if nullifier_hex.starts_with(&hex_prefix)
-                                && prefix.as_ref().len() >= min_prefix_length
-                            {
-                                matching_prefixes.push(prefix.clone());
+                        // Skip invalid nullifier data instead of panicking
```

Is it correct to swallow errors? Why not use `?` or early return an `Err`?

---

Fixed - now propagating errors with descriptive messages instead of silently skipping.